### ISSUE TITLE
[CRITICAL] Dashboard binds to `0.0.0.0` with no auth, exposing training data and state-mutation endpoints to the entire LAN

### ISSUE THREAD

**Issue by MatoTeziTanka** [iverbovoy/pgolf-dashboard#2]
Paired with #1 but listing it separately because the fix is independent.

## The bug

`dashboard.py:1213`:

```python
uvicorn.run(app, host="0.0.0.0", port=args.port, log_level="warning")
```

The dashboard binds to **all interfaces** by default, with no `--host` flag to override. There's also no authentication, no token, no CORS policy, no CSRF header requirement — the server is completely open.

## Attack surface

Anyone who can reach `<dashboard-ip>:8050` can:

1. **Read raw training logs** via `GET /api/log` (L1165-1176) — returns the full remote log text as plain text, including any debug lines, file paths, hyperparameters, GPU identifiers, and the SSH connection string echoed in the "live (SSH)" run name.
2. **Read all run state** via `GET /api/data` (L297) — returns `_runs` and `_config` including the resolved remote log path.
3. **Execute arbitrary shell on the operator's laptop** via the `?ssh=` query param (see #1).
4. **Delete runs from dashboard state** via `GET /api/remove?key=...` (L1159) — state loss, not destructive, but no auth either.
5. **Upload arbitrary "log" content** via `POST /api/upload` (L315) — gets parsed and displayed, which matters because display uses `innerHTML` with unescaped content (separate XSS issue coming).

## Real-world blast radius

- **Coffee shop WiFi**: anyone on the same network sees your training runs and can trigger RCE.
- **Corporate LAN**: other employees can watch your training progress and potentially exfiltrate logs that contain paths, hostnames, or embedded credentials.
- **Home network with UPnP misconfiguration**: potentially visible from the public internet.
- **RunPod pod with direct internet ingress**: if someone runs the dashboard ON the pod itself, port 8050 is exposed to the RunPod public IP.

## Fix

1. **Default to `127.0.0.1`.** Add a `--host` flag with a default of `"127.0.0.1"` and accept `"0.0.0.0"` only when explicitly set.
2. **Loud warning in the README and stdout when `--host 0.0.0.0` is used**, explaining the risk.
3. **Add auth if remote binding is needed.** Generate a random token on startup, print it to stdout once, require it as a `?token=...` query param or `Authorization: Bearer <token>` header on every API endpoint. Rotate per process.
4. **Add a minimal CORS policy** that rejects cross-origin requests except from `http://localhost:{port}` / `http://127.0.0.1:{port}`.
5. **Use POST for state-mutation endpoints** (`/api/data?ssh=`, `/api/remove`). GET should be read-only.

## Suggested patch (minimal)

```python
# L1183
parser.add_argument("--host", default="127.0.0.1",
                    help="Bind address (default: 127.0.0.1; use 0.0.0.0 to expose — NOT RECOMMENDED without auth)")
# L1213
if args.host == "0.0.0.0":
    print("\n⚠  Dashboard bound to 0.0.0.0 — anyone on your network can read logs and execute SSH commands.")
    print("   Add --host 127.0.0.1 to restrict to this machine.\n")
uvicorn.run(app, host=args.host, port=args.port, log_level="warning")
```

That's the bare minimum. Proper fix is token-based auth on all endpoints.

---

*Review methodology: read the file at the initial commit, verified line numbers independently, dual-confirmed with a second audit pass.*

*Disclosure: I use Claude Code CLI, Codex CLI, and Gemini Pro as tools in my workflow. Human first, AI-assisted.*
